LESS to SASS (SCSS)

foxted opened the pull request:
Converted LESS files to SASS (SCSS) as requested in #126. 

Tasks: 

- [x] Converted every LESS file into SASS
- [x] Load bootstrap & font-awesome via NPM, added to dependencies in `package.json`, so no need to include them in the project anymore (`resources/assets/sass`)
- [x] Consolidated vendor CSS into the core
- [x] Consolidated original theme into the core (admin)
- [x] Remove `auth.css`, consolidated into `backend.css`
- [x] Simplify file structure for easy customization
- [x] Font Awesome & Prism kept as CDN for the moment

Also, I checked that the conversion didn't affect anything, I didn't notice any quirks, feel free to QA again to make sure everything is there, if not, let me know, I'll update the code accordingly.

austintoddj commented:
@foxted Thanks so much for tackling this, it was a beast I didn't want to face just yet. I'll check out your changes and let you know if I see anything that needs tweaking before merging in. And as far as your proposed changes, I'm **all for it** as long as the UI all remains in-tact and functioning as it does now. So any way that you'd like to structure, re-structure, or modify that portion, please feel free. 

The last thing I'd ask is that we have a way to have *child-theme files*. This would make it so that anyone who uses the project and wants to modify the styles/JS, could do it, and then safely move that theme to any new version of Canvas that comes out. Does that make sense? So I believe in reality it would only mean having 1, 2 or however many SCSS files just blank, waiting for people if they want to write in there, and 1 JS file if they want to add any custom scripts.

austintoddj commented:
Ok, checked it out and it looks like it's working great. I'll merge it in. 

If you keep checking items off that list refactoring, I'll pull them in as well. As a thought, here is an example of a project I had that I wonder if it'd be a good way to structure the SASS directory for this project:
![screen shot 2016-08-17 at 7 35 07 am]
(https://cloud.githubusercontent.com/assets/8037011/17736420/5c5de2f6-644d-11e6-9d07-77241855e3c3.png)

Again, that's just a suggestion, but I'm open for however you want to re-work it and make it better. 👍

foxted commented:
Refactored a lot of the SCSS code.

Created two main stylesheets (entry-points): 

- `frontend.scss`
- `backend.scss`

It seemed straightforward to me, to follow the folder structure of the views. I also dropped the specific `auth.css`, this is included in the `backend.css` (it was not loaded anywhere in the views anyway ;))

Wild guess: did you use a theme that you bought somewhere (theme forest)? There is a lot of CSS that does not seem to be used, but filter through the whole codebase might take a while! I am suggesting we consolidate it into the core, minified. (In the latest commit).

Also, for now, I think we can keep Font Awesome & Prism coming from CDN. 

My own QA shows no changes in the UI, but of course, don't forget to double check!
